lib/assertions.js

```diff
@@ -68,17 +68,20 @@ exports.props = {
       }
     },
 
-    BUTTON_ROLE_NO_KEYUP: {
-      msg: 'You have `role="button"` but did not define an `onKeyUp` handler. Add it, and have the "Space" key do the same thing as an `onClick` handler.',
+    // onKeyUp is too late to cancel space's default behavior of scrolling the
+    // page.
+    BUTTON_ROLE_SPACE: {
+      msg: 'You have `role="button"` but did not define an `onKeyDown` handler. Add it, and have the "Space" key do the same thing as an `onClick` handler.',
       test (tagName, props, children) {
-        return !(props.role === 'button' && !props.onKeyUp);
```

Comment:
Remove onKeyUp. When you study what the browser does, onKeyUp is fired after the click event. So, if you want the Spacebar and Enter key to trigger a click, you'd do that onKeyDown.